### ISSUE TITLE
False positives when specifying MD003 or MD001 styles

### ISSUE THREAD

**Issue by GrayedFox** [markdownlint/markdownlint#159]
If I leave the default values (consistent) for this sample file, I get no output (clean) when running MDL on the following file (all rules enabled, own styles commented out):

```
# Documentation Header

## Introduction

Welcome to documentation about everything! Yay!

- The setup, installation, and contributing instructions for a specific service should **remain in that services README file*, but a link to that repository should also exist here
- Any team specific processes - private or otherwise - should be kept in that team's Confluence page, but a link to that team's space should also exist here
- Sections are broken down into Services and Processes
  - A process is something we do that reflects company policies or best practises such as onboarding, security measures, office management, etc
  - A service is a project contained in a specific repository (normally owned by a single team, but not necessarily), which is tasked with doing a particular thing (such as Ikaros, Aphrodite, Athena, etc)

## Contributing

Please follow the style guide enforced by our [Markdown Linter](https://github.com/mivok/markdownlint). You could also adjust them and make a PR if you want to see a style change. Check out the full list of rules [here](https://github.com/mivok/markdownlint/blob/master/docs/RULES.md).

## Services

### Foo

- [FooService](FooService)

## Processes

### Bar

- [BarProcess](BarProcess)
```

However by enabling rules MD003 and MD001 (list style and header style) which match the sample file's syntax, suddenly I get errors. My lintingStyle.rb:

```
all # all rules enabled by default

rule 'MD003', :style => "atx" # header style should be atx
rule 'MD004', :style => "dash" # list style should always use dashes

exclude_rule 'MD013' # line length can be anything, since GitHub wraps it for us anyway
```

This will produce the following output:
```
/Users/cfisher/Desktop/DocumentatioExample.md:1: MD003 Header style
/Users/cfisher/Desktop/DocumentatioExample.md:3: MD003 Header style
/Users/cfisher/Desktop/DocumentatioExample.md:13: MD003 Header style
/Users/cfisher/Desktop/DocumentatioExample.md:17: MD003 Header style
/Users/cfisher/Desktop/DocumentatioExample.md:19: MD003 Header style
/Users/cfisher/Desktop/DocumentatioExample.md:23: MD003 Header style
/Users/cfisher/Desktop/DocumentatioExample.md:25: MD003 Header style
/Users/cfisher/Desktop/DocumentatioExample.md:7: MD004 Unordered list style
/Users/cfisher/Desktop/DocumentatioExample.md:8: MD004 Unordered list style
/Users/cfisher/Desktop/DocumentatioExample.md:9: MD004 Unordered list style
/Users/cfisher/Desktop/DocumentatioExample.md:10: MD004 Unordered list style
/Users/cfisher/Desktop/DocumentatioExample.md:11: MD004 Unordered list style
/Users/cfisher/Desktop/DocumentatioExample.md:21: MD004 Unordered list style
/Users/cfisher/Desktop/DocumentatioExample.md:27: MD004 Unordered list style
```

**Comment by deivid-rodriguez** [comment]
There are several problems here:

* This should probably be more flexible, strings should be allowed too.
* The word "style" is confusing since it's used for both "code styles" (your `lintingStyle.rb`), and "rule styles" (the `atx` parameter to the rule).
* "Code styles" accept strings via `.mdlrc` but "rule styles" need to be symbols.

Anyways, your problem should be fixed by replacing `"atx"` and `"dash"` with `:atx` and `:dash` in your `lintingSyle.rb` file.

**Comment by GrayedFox** [comment]
Thank you! Will report back results today or tomorrow!

**Comment by dkniffin** [comment]
I also ran into this problem. The root cause is [the docs](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md) make it seem like the param options are strings, when they are in fact symbols.

**Comment by NicolasWebDev** [comment]
I am trying to set setext_with_atx myself, and the style: :setext_with_atx is not parsed from my .mdlrc. It is set to the default :consistent value.
